indexer/services/comlink/src/controllers/api/v4/social-trading-controller.ts

```diff
@@ -0,0 +1,87 @@
+import {
+  SubaccountTable,
+  SubaccountUsernamesTable,
+} from '@dydxprotocol-indexer/postgres';
+import express from 'express';
+import {
+  checkSchema,
+  matchedData,
+} from 'express-validator';
+
+import { getReqRateLimiter } from '../../../caches/rate-limiters';
+import { NotFoundError } from '../../../lib/errors';
+import { handleControllerError } from '../../../lib/helpers';
+import { rateLimiterMiddleware } from '../../../lib/rate-limit';
+import { handleValidationErrors } from '../../../request-helpers/error-handler';
+import ExportResponseCodeStats from '../../../request-helpers/export-response-code-stats';
+import { SubaccountInfoToTraderSearchResponse } from '../../../request-helpers/request-transformer';
+
+const router: express.Router = express.Router();
+const controllerName: string = 'social-trading-controller';
+
+function checkIfValidDydxAddress(address: string): boolean {
+  const pattern = /^dydx[0-9a-z]{39}$/;
+  return pattern.test(address);
+}
+
+async function searchTrader(
```

Comment:
Use the controller framework that we use in other controllers with tsoa so we get autogenerated documentation